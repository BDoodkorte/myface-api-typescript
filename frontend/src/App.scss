@import "./UserDetail/UserDetail.scss";
@import "./UserList/UserList.scss";

body {
  font-family: "Segoe UI", Helvetica, sans-serif;
  color: #ffffff;
  margin: 0;
}

.buttons {
  border-radius: 10px;
  border-color: #ffffff;
  border-style: solid;
  background-color: #151820;
  width: 50%;
}
