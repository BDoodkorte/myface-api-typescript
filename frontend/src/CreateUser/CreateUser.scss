

.create-user-form {
    display: grid;
    row-gap: 5px;
    width: 35%;
    min-width: 290px;
    border-radius: 10px;
    border-color: rgb(255, 255, 255);
    border-style: solid;
    background: #151820;
    padding: 8%;
}

.create-user-input {
    grid-column-start: 2;
    grid-column-end: 3;

}

.create-user-submit {
    display: grid;
    grid-column-start: 1;
    grid-column-end: 3;
    height: 35px;
    align-content: center;
    border-radius: 10px;
    border-color: #ffffff;
    border-style: solid;
    background-color: #151820;
    margin-right: 5%;

}

label {
    grid-column-start: 1;
    grid-column-end: 1;

}
