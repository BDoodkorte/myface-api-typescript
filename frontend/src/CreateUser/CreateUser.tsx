import React from "react";  // import React (to provide access to TSX)
import { useState, useEffect, FormEvent, } from 'react'
import { UserModel } from '../../../src/models/api/userModel'
import { Page } from '../../../src/models/api/page'
import { BrowserRouter as Router, Routes, Route, Link, redirect } from 'react-router-dom';
import { isJSDocLinkLike } from "typescript";
import { createUser } from "../APIBackend/APIBackend";
import { useNavigate } from "react-router-dom";
import { MenuBar } from "../Menu/Menu";

export function CreateUser() {
    const [name, setName] = useState<string>("");
    const [userName, setuserName] = useState<string>("");
    const [email, setEmail] = useState<string>("");
    const [profileImageUrl, setProfileImageUrl] = useState<string>("");
    const [coverImageUrl, setCoverImageUrl] = useState<string>("");
    const [error, setError] = useState<string>("");


    const navigate = useNavigate();

    function handleSubmit(event: FormEvent<HTMLFormElement>) {
        event.preventDefault()
        createUser(name, userName, email, profileImageUrl, coverImageUrl)
            .then(() => {
                alert("Welcome to MyFace!")
                navigate("/users")
            })
            .catch((e) => setError(e.message))


    }

    return (
        <div>

            {/* <div className="navigationButtonGroup">
                <button type="button" className="navigationbutton" onClick={() => navigate("/users")} >Users List</button>
                <button type="button" className="navigationbutton" onClick={() => navigate("/posts")} >Post List</button>
                <button type="button" className="navigationbutton" onClick={() => navigate("/posts/create")} >Create Post</button>
            </div> */}
            <h1>Create User</h1>
            <p>{error}</p>
            <form className="create-user-form" onSubmit={(e) => {
                handleSubmit(e)
            }
            }>
                <label>
                    Name:
                </label>
                <input className="create-user-input "
                    type="text"
                    name="name"
                    onChange={event => setName(event.target.value)}
                />

                <br />
                <p></p>
                <label>
                    Username:
                </label>
                <input className="create-user-input "
                    type="text"
                    name="username"
                    onChange={event => setuserName(event.target.value)}
                />

                <br />
                <p></p>
                <label>
                    Email:
                </label>
                <input className="create-user-input "
                    type="email"
                    name="email"
                    onChange={event => setEmail(event.target.value)}
                />

                <br />
                <p></p>
                <label >
                    ProfileImageUrl:
                </label>
                <input className="create-user-input "
                    type="text"
                    name="profileImageUrl"
                    onChange={event => setProfileImageUrl(event.target.value)}
                />

                <br />
                <p></p>
                <label>
                    CoverImageUrl:
                </label>
                <input className="create-user-input "
                    type="text"
                    name="coverImageUrl"
                    onChange={event => setCoverImageUrl(event.target.value)}
                />

                <br />
                <p></p>
                <button className="create-user-submit" type="submit">Submit</button>
            </form>
        </div>

    )
}

