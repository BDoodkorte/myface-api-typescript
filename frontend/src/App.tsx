import { useState, useEffect } from 'react'
import './App.scss'
import { PostList } from './PostList/PostList'
import { UserList } from './UserList/UserList';
import { UserDetail } from './UserDetail/UserDetail';
import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';
import { CreateUser } from './CreateUser/CreateUser';
import { CreatePost } from './CreatePost/CreatePost';

function App() {


  return (

    <Router>
      <Routes>
        <Route path='/posts'
          element={<PostList />}
        />
        <Route path='/users/'
          element={<UserList />}
        />
        <Route path='/users/:userId/'
          element={<UserDetail />}
        />
        <Route path='/users/create'
          element={<CreateUser />}
        />
        <Route path='/posts/create'
          element={<CreatePost />}
        />      </Routes>
    </Router>

  );
}

export default App
/* <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </div> */