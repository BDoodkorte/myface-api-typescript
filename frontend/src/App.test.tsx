import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { CreateUser } from './CreateUser/CreateUser';
import { create } from 'react-test-renderer';
import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';

test('renders MyFace text at least once', () => {
  const { getAllByText } = render(<App />);
  const textElement = getAllByText(/MyFace/i);
  expect(textElement[0]).toBeInTheDocument();
});

describe("My Component", () => {
    it("Test that snapshot of CreateUser component matches previous snapshot", async () => {
      const tree = create(<Router><CreateUser /></Router>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
