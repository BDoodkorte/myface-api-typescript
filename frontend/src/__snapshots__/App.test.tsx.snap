// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Test that snapshot of CreateUser component matches previous snapshot 1`] = `
<div>
  <h1>
    Create User
  </h1>
  <p />
  <form
    className="create-user-form"
    onSubmit={[Function]}
  >
    <label>
      Name:
    </label>
    <input
      className="create-user-input "
      name="name"
      onChange={[Function]}
      type="text"
    />
    <br />
    <p />
    <label>
      Username:
    </label>
    <input
      className="create-user-input "
      name="username"
      onChange={[Function]}
      type="text"
    />
    <br />
    <p />
    <label>
      Email:
    </label>
    <input
      className="create-user-input "
      name="email"
      onChange={[Function]}
      type="email"
    />
    <br />
    <p />
    <label>
      ProfileImageUrl:
    </label>
    <input
      className="create-user-input "
      name="profileImageUrl"
      onChange={[Function]}
      type="text"
    />
    <br />
    <p />
    <label>
      CoverImageUrl:
    </label>
    <input
      className="create-user-input "
      name="coverImageUrl"
      onChange={[Function]}
      type="text"
    />
    <br />
    <p />
    <button
      className="create-user-submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
